src/middleware/mod.rs

```diff
@@ -166,6 +168,7 @@ pub trait Pod: fmt::Debug + DynClone {
     }
     // Used for downcasting
     fn into_any(self: Box<Self>) -> Box<dyn Any>;
+    fn serialized_proof(&self) -> String;
```

Comment:
How would you see if this method does not belong to the middleware nor backend, and it's done at the frontend level only?
And the specific 'proof' serialization would be then done at the frontend at: https://github.com/0xPARC/pod2/tree/4ac78347c19339cb4080c032922a3bef28be681f/src/frontend/serialization.rs#L96

In a similar way that for deserialization it is done at the frontend, and not at the backend: 
https://github.com/0xPARC/pod2/tree/4ac78347c19339cb4080c032922a3bef28be681f/src/frontend/serialization.rs#L79 .